Dispatcher system

bill-fitch03 opened the pull request.

danielb987 commented:
@dsand47 
Do you have time to take a look at this PR? It looks good to me, but I have never used Dispatcher system.

dsand47 reviewed (dismissed):
A minimal test ran without errors.

danielb987 commented:
@bill-fitch03 
Is it time to review this PR?

bill-fitch03 commented:
I have just rewritten the help and added a few more changes, after doing some more testing with actual trains. I will submit final version in a couple of days after final checks.

bill-fitch03 commented:
Should be OK to review. Tested with two real trains running under scheduler.

bill-fitch03 commented:
Would be nice if the throttle did not drop out so frequently when dispatcher runs, but have changed code in simulation to cope with this.

danielb987 commented on java/src/jmri/jmrit/dispatcher/DispatcherFrame.java:
```diff
@@ -1451,7 +1456,7 @@ public ActiveTrain createActiveTrain(String transitID, String trainID, int tSour
         }
         if (_HasOccupancyDetection && (!(startBlock.getState() == Block.OCCUPIED))) {
-            if (showErrorMessages) {
+            if (showErrorMessages && !this.dispatcherSystemSchedulingInOperation) {
```
Static analysis CI fails on this line.

Replace `this.dispatcherSystemSchedulingInOperation` with `DispatcherFrame.dispatcherSystemSchedulingInOperation`.

(Replace `this` with `DispatcherFrame`)

danielb987 commented:
@dsand47 
PR looks good to me. Do you have anything to add?

dsand47 commented:
It is probably ok.  My concern is the change to DispatcherFrame.  @Pugwash1 should review that.

The help page is a mess with 249 warnings.  Apparently the program used to build the help page is quite old.  Fortunately most browsers are able to cope with the warnings.

Pugwash1 commented:
@danielb987 I see no problem with the DispatcherFrame change.
